src/models/Vendor.js

```diff
@@ -0,0 +1,62 @@
+const mongoose = require('mongoose');
+const Schema = mongoose.Schema;
+const bcrypt = require('bcryptjs');
+const jwt = require('jsonwebtoken');
+
+const VendorSchema = new Schema({
+  name: {
+    type: String,
+    required: [true, 'Please provide the company name'],
+    maxlength: 50
+  },
+  email: {
+    type: String,
+    required: [true, 'Please provide the company email'],
+    unique: true
+  },
+  address: {
+    type: Schema.Types.ObjectId,
+    ref: 'Address'
+  },
+  contact: {
+    type: Schema.Types.ObjectId,
+    ref: 'Contact'
+  },
+  services: {
```

Comment:
this can't be right. there is no ref to Schema. I think two schemas will be involved when the vendor and/or service is registered. One Vendor can have multiple services. Service has to have a reference to the Vendor not the other way.